sw/device/tests/hmac_enc_idle_test.c

```diff
@@ -0,0 +1,147 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_clkmgr.h"
+#include "sw/device/lib/dif/dif_hmac.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/clkmgr_testutils.h"
+#include "sw/device/lib/testing/hmac_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static dif_clkmgr_t clkmgr;
+static dif_hmac_t hmac;
+
+HMAC_TESTUTILS_ASSERT_LITTLE_ENDIAN;
+static const dif_hmac_transaction_t kHmacTransactionConfig = {
+    .digest_endianness = kDifHmacEndiannessLittle,
+    .message_endianness = kDifHmacEndiannessLittle,
+};
+
+/**
+ * https://csrc.nist.gov/CSRC/media/Projects/
+ * Cryptographic-Standards-and-Guidelines/documents/examples/HMAC_SHA256.pdf
+ *
+ * Key Length: 100
+ * Tag length: 32
+ *
+ * When key is > than the block size, it should be hashed to obtain the block
+ * sized key. Please refer to:
+ * https://csrc.nist.gov/csrc/media/publications/fips/198/archive/
+ * 2002-03-06/documents/fips-198a.pdf
+ *
+ * Specifically chapter 3 and 5 (Table 1).
+ */
+
+static const char kData[34] = "Sample message for keylen=blocklen";
+
+static uint8_t kHmacLongKey[100] = {
```

Comment:
Is there a reason this isn't a more common size, like 128 bits? Why isn't this const?

Reply:
Test vector, in the example. Makes for the better test, ensures that DIF/HW can handle awkward cases.